src/hydro/hydro_cuda.cu

```diff
@@ -1207,12 +1193,12 @@ __device__ void Average_Cell_All_Fields( int i, int j, int k, int nx, int ny, in
   Average_Cell_Single_Field( 4, i, j, k, nx, ny, nz, ncells, conserved );
   #ifdef  MHD
     // Average MHD
-    Average_Cell_Single_Field( 5+NSCALARS, i,   j,   k,   nx, ny, nz, ncells, conserved );
-    Average_Cell_Single_Field( 6+NSCALARS, i,   j,   k,   nx, ny, nz, ncells, conserved );
-    Average_Cell_Single_Field( 7+NSCALARS, i,   j,   k,   nx, ny, nz, ncells, conserved );
-    Average_Cell_Single_Field( 5+NSCALARS, i-1, j,   k,   nx, ny, nz, ncells, conserved );
-    Average_Cell_Single_Field( 6+NSCALARS, i,   j-1, k,   nx, ny, nz, ncells, conserved );
-    Average_Cell_Single_Field( 7+NSCALARS, i,   j,   k-1, nx, ny, nz, ncells, conserved );
+    Average_Cell_Single_Field( grid_enum::magnetic_x, i,   j,   k,   nx, ny, nz, ncells, conserved );
```

Comment:
I'm actually curious about how this works. Does the magnetic field averaging violate the divergence rule? This may be more complicated than the way it is currently written.

Reply:
No clue. I don't think it does but I haven't checked

Reply:
I looked into this. While I'm not 100% positive I'm pretty it's fine. Similar averaging is done in other places

Reply:
I believe this is resolved so I'm marking it accordingly. Feel free to reopen if that's not the case